core/src/avm2/object/class_object.rs

```diff
@@ -908,100 +919,48 @@ impl<'gc> TObject<'gc> for ClassObject<'gc> {
             return Err(format!("Class {:?} is not generic", self_class.read().name()).into());
         }
 
-        if !self_class.read().param().is_none() {
-            return Err(format!("Class {:?} was already applied", self_class.read().name()).into());
-        }
-
         //Because `null` is a valid parameter, we have to accept values as
         //parameters instead of objects. We coerce them to objects now.
         let object_param = match nullable_param {
             Value::Null => None,
             Value::Undefined => return Err("Undefined is not a valid type parameter".into()),
```

Comment:
The case for `Value::Undefined` should be removed: `Vector.<undefined>` has the same error as `Vector.<3>` in Flash.